modules/end-user-portal/src/Core/HttpClientServices/SiteService.cs

```diff
@@ -0,0 +1,57 @@
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpClientServices.Interfaces;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Helpers;
+using Newtonsoft.Json;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpClientServices;
+
+public class SiteService : ISiteService
+{
+  private static ILogger<SiteService> _logger;
+  private readonly HttpClient _httpClient;
+  private readonly JsonSerializerSettings _options;
+
+  public SiteService(ILogger<SiteService> logger, HttpClient httpClient)
+  {
+    _logger = logger;
+    _httpClient = httpClient;
+    _options = new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore };
+  }
+
+  public async Task<TResult> SearchSitesAsync<TRequest, TResult>(TRequest searchRequest) where TRequest : class where TResult : class
```

Comment:
This doesn't need to be generic any more we know that we will send a SearchRequest and get back a SiteResults object

Reply:
Changed to be non generic